### PR TITLE
Launch: Move launch store to package/data-stores.

### PR DESCRIPTION
**Pull request by yansern**
#### Changes proposed in this Pull Request

* Move launch store to `package/data-stores`.
* Also fixed several type issues with launch stores.

#### Testing instructions

* Run `yarn start`.
* Run `yarn dev --sync` on `apps/editing-toolkit`.
* Test on sandbox.
* Launch flow should work as usual.
  * Also test for data persistence, values like selected domain, searched domain keyword, and selected plan should be remembered.
* Additionally, test gutenboarding `/new` for any possible regression (unlikely but hey).

### CODE REVIEW

**Review by wp-desktop — DISMISSED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@yansern please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/927490). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

File: packages/data-stores/src/launch/actions.ts [hunk 1]
```diff
@@ -101,4 +102,6 @@ export type LaunchAction = ReturnType<
 	| typeof openSidebar
 	| typeof closeSidebar
 	| typeof enableExperimental
+	| typeof setSidebarFullscreen
```
Comment (ciampo) [inline packages/data-stores/src/launch/actions.ts:105]: Types were missing in the exported values

File: packages/data-stores/src/launch/selectors.ts [hunk 2]
```diff
@@ -8,28 +8,32 @@ import { select } from '@wordpress/data';
...
 // Completion status of steps is derived from the state of the launch flow
-export const isStepCompleted = ( state: State, step: LaunchStepType ) => {
+export const isStepCompleted = ( state: State, step: LaunchStepType ): boolean => {
 	if ( step === LaunchStep.Plan ) {
 		return !! getSelectedPlan( state );
 	}
 	if ( step === LaunchStep.Name ) {
-		const site = select( 'core' ).getEntityRecord( 'root', 'site', undefined );
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		const site: any = select( 'core' ).getEntityRecord( 'root', 'site', undefined );
```
Comment (ciampo) [inline packages/data-stores/src/launch/selectors.ts:36]: Maybe we can (partially) type the `site` variable as `{ site?: 'string' } | undefined`

**Review by ciampo — CHANGES_REQUESTED** [review]

**Review by ciampo — APPROVED** [review]
Tested changes, everything works as before the move to different package

**Pull request merged**